### PR TITLE
[SDK-2445]: add support to register multiple push primer listeners

### PR DESCRIPTION
**Pull request by shivamsharma2710**

### CODE REVIEW

File: clevertap-core/src/main/java/com/clevertap/android/sdk/CleverTapAPI.java [hunk 1]
```diff
@@ -1636,14 +1637,11 @@ public void setInAppNotificationListener(InAppNotificationListener inAppNotifica
...
-    /**
-     * Returns the PushPermissionNotificationResponseListener object
-     *
-     * @return An {@link PushPermissionResponseListener} object
-     */
-    @SuppressWarnings({"unused", "WeakerAccess"})
-    public PushPermissionResponseListener getPushPermissionNotificationResponseListener() {
-        return coreState.getCallbackManager().getPushPermissionResponseListener();
+    @SuppressWarnings({"unused"})
+    public void unregisterPushPermissionNotificationResponseListener(PushPermissionResponseListener
```
Comment (darshanclevertap) [inline clevertap-core/src/main/java/com/clevertap/android/sdk/CleverTapAPI.java:1641]: @shivamsharma2710 please add javadoc for the public methods

Comment (shivamsharma2710) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/CleverTapAPI.java:1649]: This is done.

**Review by piyush-kukadiya — APPROVED** [review]
LGTM

File: clevertap-core/src/main/java/com/clevertap/android/sdk/CleverTapAPI.java [hunk 2]
```diff
@@ -1637,25 +1638,36 @@ public void setInAppNotificationListener(InAppNotificationListener inAppNotifica
...
+        coreState.getCallbackManager().
+                unregisterPushPermissionResponseListener(pushPermissionResponseListener);
     }
 
     /**
-     * This method sets the PushPermissionNotificationResponseListener
+     * This method registers the PushPermissionNotificationResponseListener.
+     * <p>
+     * Make sure to unregister the listener in onDestroy() or onPause() of the activity/fragment if the
+     * listener is registered in onCreate() or onResume() respectively.
```
Comment (darshanclevertap) [inline clevertap-core/src/main/java/com/clevertap/android/sdk/CleverTapAPI.java:1659]: @shivamsharma2710 Please mention to call this register method only in `onCreate`

**Review by darshanclevertap — COMMENTED** [review]
@shivamsharma2710 Please mention to call this register method only in `onCreate` method

**Review by darshanclevertap — APPROVED** [review]

**Pull request merged**